applications/asset_tracker/src/main.c

```diff
@@ -1756,23 +1751,17 @@ void main(void)
 
 	work_init();
 
-lte_connect:
-		ret = modem_configure();
-		if (ret) {
-			LOG_WRN("Failed to establish LTE connection.");
-			LOG_WRN("Will retry in %d seconds.",
-					CONFIG_CLOUD_CONNECT_RETRY_DELAY);
-			k_sleep(K_SECONDS(CONFIG_CLOUD_CONNECT_RETRY_DELAY));
-			goto lte_connect;
-		}
+	while (modem_configure() != 0) {
+		LOG_WRN("Failed to establish LTE connection.");
+		LOG_WRN("Will retry in %d seconds.",
+				CONFIG_CLOUD_CONNECT_RETRY_DELAY);
+		k_sleep(K_SECONDS(CONFIG_CLOUD_CONNECT_RETRY_DELAY));
+	}
 
 #if defined(CONFIG_LWM2M_CARRIER)
 	LOG_INF("Waiting for LWM2M carrier to complete initialization...");
 	k_sem_take(&cloud_ready_to_connect, K_FOREVER);
 #endif
 
-	/* Handle connection to the cloud.
-	 * Function returns only on error or pending reboot.
-	 */
-	cloud_connection_handler();
+	connect_to_cloud(K_NO_WAIT);
```

Comment:
I wouldn't use K_NO_WAIT here, this symbol became a struct in upstream Zephyr, just use 0.